pkg/scheduler/framework/plugins/interpodaffinity/filtering_test.go

```diff
@@ -988,21 +985,24 @@ func TestPreFilterDisabled(t *testing.T) {
 	cycleState := framework.NewCycleState()
 	gotStatus := p.(framework.FilterPlugin).Filter(context.Background(), cycleState, pod, nodeInfo)
 	wantStatus := framework.AsStatus(fmt.Errorf(`error reading "PreFilterInterPodAffinity" from cycleState: %w`, framework.ErrNotFound))
-	if !reflect.DeepEqual(gotStatus, wantStatus) {
-		t.Errorf("status does not match: %v, want: %v", gotStatus, wantStatus)
+	if wantStatus.Code() != gotStatus.Code() {
```

Comment:
Compare of two status is always a problem to us.

Maybe this func can help instead of comparing each field in tests(Not sure).
```
// Equal checks equality of two statuses. This is useful for testing with
// cmp.Equal.
func (s *Status) Equal(x *Status) bool {
```

Reply:
Also, seems we defined some `cmpopts` for comparing of two status. 
```
var cmpOpts = []cmp.Option{
	cmp.Comparer(func(s1 *framework.Status, s2 *framework.Status) bool {
		if s1 == nil || s2 == nil {
			return s1.IsSuccess() && s2.IsSuccess()
		}
		return s1.Code() == s2.Code() && s1.Plugin() == s2.Plugin() && s1.Message() == s2.Message()
	}),
}
```
Maybe it's time to unify this behavior? 
WDYT @kerthcet @sanposhiho 
@sivchari Let's wait other reviewers to reach an agreement on this.

Reply:
We can just use cmp.Diff, can't we?
`cmp.Diff` takes `Equal` function into consideration:
> Types with an Equal method (e.g., [time.Time.Equal](https://pkg.go.dev/time#Time.Equal)) may use that method to determine equality. This allows package authors to determine the equality operation for the types that they define.
https://pkg.go.dev/github.com/google/go-cmp/cmp

Reply:
Hmmm, from the doc it seems yes. However, when I try to remove `opts`, I got an error:
`FAIL: TestFilterPluginsWithNominatedPods/node_has_a_high-priority_nominated_pod_and_filters_fail `
Maybe it's something wrong in our `Equal` method🤦.

Reply:
We should modify `Equal` method then, rather than having `opts` every places where we're using `cmp.Diff`.

Reply:
Yep, in `Equal` method, we compare errors of two status, which will not be true in most cases.
```
	if !cmp.Equal(s.err, x.err, cmpopts.EquateErrors()) {
		return false
	}
```
What's your suggestion? Should we just compare `err.ERROR()` here?

Reply:
We should put the root error in test cases. 

Supposing we have this function, which we want to test:

```go
var ErrNotFound = errors.New("not found")

func Hoge() *framework.Status {
...
return framework.AsStatus(fmt.Errorf("hogehoge goes wrong: %w", ErrNotFound))
}
```

Then, in the test case, we can set `framework.AsStatus(ErrNotFound)` to `wantStatus`. 
I believe then `Equal` method works as it is.

Reply:
Some functions may not have root errors like the above `ErrNotFound` though, then we should modify the implementation to have root errors. That's the ideal state.